CI/macos/appcast_download.py

```diff
@@ -0,0 +1,62 @@
+import os
+import sys
+
+import requests
+import xmltodict
+
+
+def download_build(url):
+    print(f'Downloading build "{url}"...')
+    filename = url.rpartition("/")[2]
+    r = requests.get(url)
+    if r.status_code == 200:
+        with open(f"artifacts/{filename}", "wb") as f:
+            f.write(r.content)
+    else:
+        print(f"Build download failed, status code: {r.status_code}")
+        sys.exit(1)
+
+
+def read_appcast(url, fallback):
+    r = requests.get(url)
+    if r.status_code == 404:
+        print("Using fallback Appcast URL...")
+
+        r = requests.get(fallback)
+        if r.status_code != 200:
+            print(f"Appcast download failed, status code: {r.status_code}")
+            sys.exit(1)
+
+    filename = url.rpartition("/")[2]
+    with open(f"builds/{filename}", "wb") as f:
+        f.write(r.content)
+
+    appcast = xmltodict.parse(r.content, force_list=("item",))
+
+    dl = 0
+    for item in appcast["rss"]["channel"]["item"]:
+        channel = item.get("sparkle:channel", "stable")
+        if channel != target_branch:
+            continue
+
+        if dl == max_old_vers:
+            break
+        download_build(item["enclosure"]["@url"])
+        dl += 1
+
+
+if __name__ == "__main__":
+    appcast_urls = {
+        "x86_64": (
+            "https://obsproject.com/osx_update/updates_x86_64_v2.xml",
+            "https://obsproject.com/osx_update/stable/updates_x86_64.xml",
+        ),
+        "arm64": (
+            "https://obsproject.com/osx_update/updates_arm64_v2.xml",
+            "https://obsproject.com/osx_update/stable/updates_arm64.xml",
+        ),
+    }
+    target_branch = os.getenv("BRANCH")
```

Comment:
As always, I _really_ don't like relying on environment variables - if these are parameters for the script they should be parsed and handled as such.

Reply:
Yeah that could just be a simple bit of argparse, didn't really think much of it at the time since it's only ever meant to run on CI.

Reply:
It's not a blocker IMO. I just prefer any code (even little scripts) to be fully-fledged and production-quality.